src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs

```diff
@@ -1015,6 +1015,296 @@ public void GetAsync_ServerNeedsAuthAndNoCredential_StatusCodeUnauthorized()
             }
         }
 
+        [Theory]
+        [InlineData(":")]
+        [InlineData("  :  ")]
+        [InlineData("\x1234: \x5678")]
+        [InlineData("nocolon")]
+        [InlineData("no colon")]
+        [InlineData("Content-Length      ")]
+        public async Task GetAsync_InvalidHeaderNameValue_ThrowsHttpRequestException(string invalidHeader)
```

Comment:
Should we move these header tests to HttpProcotolTests.cs?  That was my intention in creating the file; unfortunately I only got as far as some basic tests re status line and protocol versions.